components/script/dom/readablestream.rs

```diff
@@ -427,73 +464,142 @@ impl ExternalUnderlyingSourceController {
         None
     }
 
+    fn add_reader(&self, cx: SafeJSContext, promise: Rc<Promise>) {
+        assert!(self.pending_reader.borrow().is_none());
+        *self.pending_reader.borrow_mut() = Some(promise.clone());
+        let buflen = self.buffer.borrow().len();
+        self.maybe_signal_available_bytes(cx, &promise.global(), buflen);
```

Comment:
The problem with this approach is that it will only wake up the reader in reponse to `enqueue_native`, and in practice the queuing could also be done by a JS underlying source([example](https://streams.spec.whatwg.org/#ref-for-push-source%E2%91%A1)).

That was the part that SM dealt with internally, I think.